### PR TITLE
Editable kernel arguments

### PR DESCRIPTION
**Pull request by matbme**
While setting up Vanilla OS on my laptop, I found the need to add extra kernel parameters, but quickly realized that those parameters get replaced every time the root partition updates. This PR allows ABroot to read a string containing kernel parameters from `/etc/abroot/kargs` while updating the root partitions.

The only check I implemented so far is to make sure the string read from `kargs` doesn't end with a newline (which would break the config), but given the potential to make the system "unbootable", maybe it's a good idea to check the format as well. I think it's also a good idea to create a new file with the default parameters if it cannot be found, which would prevent the need for manual intervention if this gets merged. Please let me know your intentions so I can add this if you plan to merge this PR :)

### CODE REVIEW

**Comment by mirkobrombin** [comment]
I had planned to work on this and had the same concept in mind. Thanks for the contribution!

I think we should consider not updating the kargs of the present partition as if the user makes errors, they will be reflected on both partitions, preventing the system from boot. My idea is to duplicate kargs on transaction start if it exists and use that only for the partition present.

Concerning the standard flags, we could define a kargs.default in /etc/abroot and install it together with ABRoot.

**Comment by matbme** [comment]
> I think we should consider not updating the kargs of the present partition as if the user makes errors, they will be reflected on both partitions

Makes sense, I'll change that.

> we could define a kargs.default in /etc/abroot and install it together with ABRoot.

Do I do this by adding a new file to `./debian/abroot.install`?

**Comment by mirkobrombin** [comment]
> Do I do this by adding a new file to ./debian/abroot.install?

adding it in /debian/etc/abroot/kargs.default should works

**Comment by matbme** [comment]
ABroot now tries to read `/etc/abroot/kargs.default` if `/etc/abroot/kargs` can't be read.

File: core/transaction.go [hunk 1]
```diff
@@ -8,9 +8,11 @@ import (
 )
 
 var (
-	lockPath       = "/tmp/abroot-transactions.lock"
-	startRulesPath = "/etc/abroot/start-transaction-rules.d/"
-	endRulesPath   = "/etc/abroot/end-transaction-rules.d/"
+	lockPath            = "/tmp/abroot-transactions.lock"
+	startRulesPath      = "/etc/abroot/start-transaction-rules.d/"
+    kargsPath           = "/etc/abroot/kargs"
```
Comment (pietrodicaprio) [inline core/transaction.go:13]: fix this indentation please

File: core/transaction.go [hunk 2]
```diff
@@ -136,12 +160,17 @@ func ApplyTransaction() error {
 	}
 
 	PrintVerbose("step:  UpdateRootBoot")
-	if err := UpdateRootBoot(true); err != nil {
-		_ = UnmountFutureRoot()
+    kargs, err := GetKargs()
```
Comment (pietrodicaprio) [inline core/transaction.go:163]: fix indentation please

**Review by pietrodicaprio — CHANGES_REQUESTED** [review]
See line comments

**Comment by matbme** [comment]
Classic tabs vs spaces problem 😅

**Comment by kbdharun** [comment]
> Classic tabs vs spaces problem sweat_smile

Haha, Yeah

**Comment by pietrodicaprio** [comment]
> Classic tabs vs spaces problem 😅

no problem, great job! 💯

**Pull request merged**